system/include/emscripten/val.h

```diff
@@ -348,6 +350,14 @@ namespace emscripten {
             return val(internal::_emval_new_object());
         }
 
+        static val u8string(const char* s) {
```

Comment:
Should probably use `char8_t`.

Reply:
I think that is C++10 only

Reply:
Yeah. `char8_t` needs C++20 and later. It's a distinct type for UTF-8 strings and has underlying type `unsigned char`.

According to Tom Honermann:
>https://stackoverflow.com/questions/57402464/is-c20-char8-t-the-same-as-our-old-char

Seems `char8_t` is more restrictive and may not accept literal strings without `u8` prefixed.  
We may overload it with `const char8_t*` later.

Reply:
Oh ok, weird that they added u8string before the dedicated char type.